src/Helper/Helper_Notices.php

```diff
@@ -210,4 +203,37 @@ private function html( $text, $class = 'updated' ) {
 		</div>
 		<?php
 	}
+
+	/**
+	 * If the current page belongs to Gravity PDF, add these hooks.
+	 *
+	 * @since 6.3
+	 */
+	public function maybe_gravitypdf_page() {
+		if ( \GPDFAPI::get_misc_class()->is_gfpdf_page() ) {
+			add_action( 'gform_admin_messages ', [ $this, 'remove_non_gravity_pdf_messages' ] );
+			add_action( 'gform_admin_error_messages', [ $this, 'remove_non_gravity_pdf_messages' ] );
+		}
+	}
+
+	/**
+	 * Remove all Non-Gravity PDF messages.
+	 * Due GFCommon::display_admin_message that only display one message at time.
+	 *
+	 * @param array $messages  The message to be displayed
+	 *
+	 * @return array $messages
+	 *
+	 * @since 6.3
+	 */
+	public function remove_non_gravity_pdf_messages( $messages ) {
+		foreach ( $messages as $i => $message ) {
```

Comment:
The idea is to disregard `$messages` on Gravity PDF-related pages and return `$this->notices` or `$this->errors` instead, thereby ensuring that non-Gravity PDF notices/errors won't show up on Gravity PDF pages.